packages/core/test/integer.test.ts

```diff
@@ -355,6 +356,121 @@ describe('Integer', () => {
       Integer.MAX_SAFE_VALUE.toString()
     )
   })
+
+  test('int(BigInt) should be reversed by Integer.toBigInt', () => {
+    fc.assert(
+      fc.property(
+        fc.bigInt({ max: Integer.MAX_SAFE_VALUE.toBigInt(), min: Integer.MIN_SAFE_VALUE.toBigInt() }),
+        bigint => bigint === int(bigint).toBigInt()
+      )
+    )
+  })
+
+  test('int(string) should be reversed by Integer.toString', () => {
+    fc.assert(fc.property(fc.integer().map(num => num.toString()), str => str === int(str).toString()))
+  })
+
+  test('int(number) should be reversed by Integer.toNumber', () => {
+    fc.assert(fc.property(fc.integer(), num => num === int(num).toNumber()))
+  })
+
+  test('int(Integer) should be equal Integer', () => {
+    fc.assert(fc.property(arbitraryInteger(), integer => integer === int(integer)))
+  })
+
+  test('Integer.add should be Commutative', () => {
+    fc.assert(fc.property(arbitraryInteger(), arbitraryInteger(),
+      (a, b) => a.add(b).equals(b.add(a))))
+  })
+
+  test('Integer.multiply should be Commutative', () => {
+    fc.assert(fc.property(arbitraryInteger(), arbitraryInteger(),
+      (a, b) => a.multiply(b).equals(b.multiply(a))))
+  })
+
+  test('Integer.add should be Associative', () => {
+    fc.assert(fc.property(arbitraryInteger(), arbitraryInteger(), arbitraryInteger(),
+      (a, b, c) => a.add(b.add(c)).equals(a.add(b).add(c))))
+  })
+
+  test('Integer.multiply should be Associative', () => {
+    fc.assert(fc.property(arbitraryInteger(), arbitraryInteger(), arbitraryInteger(),
+      (a, b, c) => a.multiply(b.multiply(c)).equals(a.multiply(b).multiply(c))))
+  })
+
+  test('Integer.add should be Distributive', () => {
+    fc.assert(fc.property(arbitraryInteger(), arbitraryInteger(), arbitraryInteger(),
+      (a, b, c) => a.multiply(b.add(c)).equals(a.multiply(b).add(a.multiply(c)))))
+  })
+
+  test('Integer.subtract should be Distributive', () => {
+    fc.assert(fc.property(arbitraryInteger(), arbitraryInteger(), arbitraryInteger(),
+      (a, b, c) => a.multiply(b.subtract(c)).equals(a.multiply(b).subtract(a.multiply(c)))))
+  })
+
+  describe('For Same Signal Integers', () => {
+    test('Integer.greaterThan should return true if a - b is positive', () => {
+      fc.assert(fc.property(
+        arbitrarySameSignalIntegers(),
+        ({ a, b }) => a.subtract(b).isPositive() ? a.greaterThan(b) : !a.greaterThan(b)))
+    })
+
+    test('Integer.greaterThanOrEqual should return true if a - b is positive or ZERO', () => {
+      fc.assert(fc.property(
+        arbitrarySameSignalIntegers(),
+        ({ a, b }) => a.subtract(b).isPositive() || a.subtract(b).isZero() ? a.greaterThanOrEqual(b) : !a.greaterThanOrEqual(b)))
+    })
+
+    test('Integer.equals should return true if a - b is ZERO', () => {
```

Comment:
Couldn't this test just be outside the case differentiation? This holds true regardless if the numbers have the same or a different sign. Or are there issues with overflows?

Reply:
We don't have, we could just move it to outside.